data/transactions/verify/txn.go

```diff
@@ -125,6 +128,9 @@ func Txn(s *transactions.SignedTxn, ctx Context) error {
 	if s.Txn.Src() == zeroAddress {
 		return errors.New("empty address")
 	}
+	if !proto.SupportRekeying && (s.AuthAddr != basics.Address{}) {
```

Comment:
Hm, I guess that you're doing this here since `WellFormed` is on a `transactions.Transaction` and not a `transactions.SignedTxn`, but quickly grepping through our code, it looks like we always a `SignedTxn` around when calling `WellFormed` (except maybe some tests?)... this doesn't have to happen here, but maybe we should update that in a separate PR since this really feels like a `WellFormed` check.

Reply:
(See comment above)